source/adapters/cuda/event.cpp

```diff
@@ -165,8 +166,19 @@ UR_APIEXPORT ur_result_t UR_APICALL urEventGetInfo(ur_event_handle_t hEvent,
   UrReturnHelper ReturnValue(propValueSize, pPropValue, pPropValueSizeRet);
 
   switch (propName) {
-  case UR_EVENT_INFO_COMMAND_QUEUE:
+  case UR_EVENT_INFO_COMMAND_QUEUE: {
+    // If the runtime owns the native handle, we have reference to the queue.
+    // Otherwise, the event handle comes from an interop API with no RT refs.
+    if (!hEvent->backendHasOwnership()) {
+      setErrorMessage("Command queue info cannot be queried for the event. The "
+                      "event object was created from a native event and has no "
+                      "valid reference to a command queue.",
+                      UR_RESULT_ERROR_INVALID_VALUE);
```

Comment:
Any opinions on the return code to use will be appreciated.

Additionally, I am willing to make it a non-backend specific error and simply return appropriate error code but I am unable to find such reasonable one. Also, I think this way we are able to relay the message to the user a bit better.

Reply:
It may be appropriate to just return UR_RESULT_ERROR_INVALID_VALUE but I won't be able to provide a more descriptive error message that way. Hence, adapter specific which in the SYCL Runtime is handled to report the custom error buffer accordingly.